lib/osproject_android.rb

```diff
@@ -25,25 +37,32 @@ def add_sdk!
       raise 
     end
 
+    network_handler = NetworkHandler.instance
+
     project_dir = dir
     app_dir = app_class_location.split('/')[0]
     build_gradle_dir = project_dir + '/build.gradle'
     build_gradle_app_dir = project_dir + '/' + app_dir + '/build.gradle'
+    success_mesage = ""
 
     begin 
       content = File.read(build_gradle_dir)
     rescue
       puts "File not found: " + build_gradle_dir 
-      puts "Provide first param project with directory path"
+      puts "Call CLI tool from base project directory"
+      error_track_message = "error=user called CLI from invalid directory file: " + build_gradle_dir + " not found;"
+      network_handler.send_track_from_message(os_app_id, 'android', lang, success_mesage, error_track_message)
       return
     end
 
     begin 
       content = File.read(build_gradle_app_dir)
     rescue
       puts "Directory not found: " + project_dir + '/' + app_dir 
-      puts "Provide second param Application file path directory. If no Appplication class available, OneSignal will create it at the directory provided."
+      puts "Provide --entrypoint param as Application file path directory. If no Appplication class available, OneSignal will create it at the directory provided."
       puts "Example: app/src/main/java/com/onesignal/testapplication/OneSignalApplication.java"
+      error_track_message = "error=user entered invalid Application file path: " + project_dir + '/' + app_dir + " directory not found;"
+      network_handler.send_track_from_message(os_app_id, 'android', lang, success_mesage, error_track_message)
```

Comment:
What do you think of creating a method in the `network_handler` to handle the error cases? Or would it be unnecessary since the behavior would be pretty similar?

Reply:
It's a good idea!